lib/vmpooler/providers/vsphere.rb

```diff
@@ -418,15 +416,15 @@ def set_network_device(datacenter_name, template_vm_network_device, network_name
           # Determine network device type
           # All possible device type options here: https://vdc-download.vmware.com/vmwb-repository/dcr-public/98d63b35-d822-47fe-a87a-ddefd469df06/2e3c7b58-f2bd-486e-8bb1-a75eb0640bee/doc/vim.vm.device.VirtualEthernetCard.html
           network_device =
-            if template_vm_network_device.is_a? RbVmomi::VIM::VirtualVmxnet2
```

Comment:
we need to check if there are subclasses that we also want to accept here? for example if `RbVmomi::VIM::VirtualE1000` has a subclass, `is_a?(RbVmomi::VIM::VirtualE1000)` would return `true` for the subclass too

Reply:
Those network device classes don't have any subclasses. They are the exact class for the virtual network device objects